compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -724,6 +865,16 @@ static LogicalResult createFusionGroups(TensorDimTrackingRewriter &rewriter,
     }
   });
 
+  if (collapse) {
```

Comment:
Oh, I am not sure why this is here. Now that the passes are split you could do it in a separate pass after `flow.dispatch.region` formation and before conversion to `flow.dispatch.workgroup`. You just need to look for `flow.dispatch.region` operations that have a single `linalg.generic` operation.
I suspect one reason is that `tensor.expand/collapse_shape` get hoisted out of dispatches automatically. I am fine with that for now, but pleae leave a TODO describing that this could be moved into its own pass instead of adding it to this pass.

Reply:
Right, I agree that it should be another pass. I added this as TODO and I'll do it after I come back from vacation.
Another issue would be here is to regenerate `workload_count` information.